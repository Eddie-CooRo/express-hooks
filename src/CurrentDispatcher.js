let CurrentDispatcher = {
	current: null,
};

export function setDispatcher(dispatcher) {
	CurrentDispatcher.current = dispatcher;
}

export function resolveDispatcher() {
	if (CurrentDispatcher.current) {
		return CurrentDispatcher.current;
	} else {
		throw new Error('Express hooks should be used inside of express handlers or middlewares');
	}
}
