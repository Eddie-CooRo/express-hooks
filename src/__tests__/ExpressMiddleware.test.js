import expressMiddleware from '../ExpressMiddleware';
import { resolveDispatcher } from '../CurrentDispatcher';

describe('ExpressMiddleware', () => {
	it('expressMiddleware successfuly sets the dispatcher', () => {
		expressMiddleware('Req', 'Res', () => {
			const dispatcher = resolveDispatcher();
			expect(dispatcher).toEqual({ _req: 'Req', _res: 'Res' });
		});
	});

	it('removes dispatcher after calling next', () => {
		expressMiddleware('Req', 'Res', () => {});
		expect(() => resolveDispatcher()).toThrowErrorMatchingSnapshot();
	});
});
