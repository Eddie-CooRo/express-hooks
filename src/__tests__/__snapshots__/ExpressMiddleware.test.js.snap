// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpressMiddleware removes dispatcher after calling next 1`] = `"Express hooks should be used inside of express handlers or middlewares"`;
